Aplib.Extensions/Actions/PathfinderAction.cs

```diff
@@ -0,0 +1,43 @@
+using Aplib.Core;
+using Aplib.Core.Belief.BeliefSets;
+using System;
+
+namespace Aplib.Extensions.Actions
+{
+    public class PathfinderAction<TBeliefset, TLocation> : Core.Intent.Actions.Action<TBeliefset>
+        where TBeliefset : IBeliefSet
+    {
+        public PathfinderAction(IMetadata metadata,
+            IPathfinder<TLocation> pathfinder,
+            Func<TBeliefset, TLocation> getCurrentLocation,
+            Func<TBeliefset, TLocation> getEndLocation,
+            Action<TBeliefset, TLocation> effect)
+            : base(metadata, ExecutePathFinder(pathfinder, getCurrentLocation, getEndLocation, effect))
+        {
+        }
+
+        public PathfinderAction(IPathfinder<TLocation> pathfinder,
+            Func<TBeliefset, TLocation> getCurrentLocation,
+            Func<TBeliefset, TLocation> getEndLocation,
+            Action<TBeliefset, TLocation> effect)
+            : this(new Metadata(), pathfinder, getCurrentLocation, getEndLocation, effect)
+        {
+        }
+
+        private static Action<TBeliefset> ExecutePathFinder(IPathfinder<TLocation> pathfinder,
```

Comment:
```suggestion
        private static Action<TBeliefset> TryMoveToNextLocation(IPathfinder<TLocation> pathfinder,
```

Reply:
No, I don't want to use that naming scheme. `Try...` symbols that it returns a bool and an `out` variable.

Reply:
I agree with Joachim